src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -8356,6 +8656,1228 @@ .maxstack  2
+  IL_002a:  nop
+  IL_002b:  ret
+}
+");
+            diff2.VerifyIL("C.<>c__DisplayClass0_0#2.<.ctor>b__0()", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldfld      ""int C.<>c__DisplayClass0_0#2.x""
+  IL_0006:  ldc.i4.1
+  IL_0007:  add
+  IL_0008:  ret
+}
+");
+        }
+
+        [Fact]
+        public void OutVar_InMethodBody_WithLambda()
+        {
+            var source0 = MarkedSource(@"
+public class C
+{
+    public void Method() { int _ = M(out int <N:0>x</N:0>) + M2(<N:1>() => x + 1</N:1>); }
+    static int M(out int x) => throw null;
+    static int M2(System.Func<int> x) => throw null;
+}");
+            var source1 = MarkedSource(@"
+public class C
+{
+    public void Method() { int _ = M(out int <N:0>x</N:0>) + M2(<N:1>() => x - 1</N:1>); }
+    static int M(out int x) => throw null;
+    static int M2(System.Func<int> x) => throw null;
+}");
+
+            var compilation0 = CreateCompilation(source0.Tree, options: ComSafeDebugDll);
+            var compilation1 = compilation0.WithSource(source1.Tree);
+            var compilation2 = compilation1.WithSource(source0.Tree);
+
+            var ctor0 = compilation0.GetMember<MethodSymbol>("C.Method");
+            var ctor1 = compilation1.GetMember<MethodSymbol>("C.Method");
+            var ctor2 = compilation2.GetMember<MethodSymbol>("C.Method");
+
+            var v0 = CompileAndVerify(compilation0);
+            v0.VerifyIL("C.Method", @"
+{
+  // Code size       38 (0x26)
+  .maxstack  3
+  .locals init (C.<>c__DisplayClass0_0 V_0, //CS$<>8__locals0
+                int V_1) //_
+  IL_0000:  newobj     ""C.<>c__DisplayClass0_0..ctor()""
+  IL_0005:  stloc.0
+  IL_0006:  nop
+  IL_0007:  ldloc.0
+  IL_0008:  ldflda     ""int C.<>c__DisplayClass0_0.x""
+  IL_000d:  call       ""int C.M(out int)""
+  IL_0012:  ldloc.0
+  IL_0013:  ldftn      ""int C.<>c__DisplayClass0_0.<Method>b__0()""
+  IL_0019:  newobj     ""System.Func<int>..ctor(object, System.IntPtr)""
+  IL_001e:  call       ""int C.M2(System.Func<int>)""
+  IL_0023:  add
+  IL_0024:  stloc.1
+  IL_0025:  ret
+}
+");
+            v0.VerifyIL("C.<>c__DisplayClass0_0.<Method>b__0()", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldfld      ""int C.<>c__DisplayClass0_0.x""
+  IL_0006:  ldc.i4.1
+  IL_0007:  add
+  IL_0008:  ret
+}
+");
+
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, ctor0, ctor1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            diff1.VerifySynthesizedMembers("C: {<>c__DisplayClass0_0#1}", "C.<>c__DisplayClass0_0#1: {x, <Method>b__0}");
```

Comment:
Tomas said the `#1` indicates a new class was generated, which is unexpected.
It might be an issue with the test. I'll investigate (he said to debug EncVariableSlotAllocator.TryGetPreviousClosure to figure this out)